releases/release-1.23/README.md

```diff
@@ -13,11 +13,11 @@ description: |
 #### Links
 
 * [This document](https://git.k8s.io/sig-release/releases/release-1.23/README.md)
-* [Release Team](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.23/release-team.md)
-* [Meeting Minutes](http://bit.ly/k8s122-releasemtg) (join [kubernetes-sig-release@] to receive meeting invites)
+* [Release Team][release-team@]
+* [Meeting Minutes](http://bit.ly/k8s123-releasemtg) (join [kubernetes-sig-release@] to receive meeting invites)
 * [v1.23 Release Calendar][k8s123-calendar]
 * Contact: [#sig-release] on slack, [kubernetes-release-team@] on e-mail
-* [Internal Contact Info][Internal Contact Info] (accessible only to members of [kubernetes-release-team@])
+* [Internal Contact Info][Internal Contact Info] (accessible only to members of [kubernetes-release-team@]/[release-team@])
```

Comment:
```suggestion
* [Internal Contact Info] (accessible only to members of [release-team@])
```

1. As the link is identical, there's no need to reference it twice
2. We drop the reference to `kubernetes-release-team@` as it's a deprecated list